<?xml version="1.0" encoding="UTF-8"?>
<!--This does not replace URDF, and is not an extension
of URDF.
    This is a format for representing semantic information about the robot structure.
    A URDF file must exist for this robot as well, where the joints and the links that are referenced
are defined
-->
<robot name="bqr3_arm">
    <group name="whole_body">
        <joint name="root_joint" />
        <joint name="shoulder_joint" />
        <joint name="upper_arm_joint" />
        <joint name="forearm_joint" />
        <joint name="wrist_pitch_joint" />
        <joint name="wrist_yall_joint" />
    </group>
    <!--GROUP
    STATES: Purpose: Define a named state for a particular group, in terms of joint values. This is
    useful to define states like 'folded arms'-->
    <group_state name="standing" group="whole_body">
        <joint name="root_joint" value="0. 0. 0.57 0. 0. 0. 1." />
        <joint name="shoulder_joint" value="0"/>
        <joint name="upper_arm_joint" value="-1.57"/>
        <joint name="forearm_joint" value="2.88"/>
        <joint name="wrist_pitch_joint" value="0.26"/>
        <joint name="wrist_yall_joint" value="0"/>
    </group_state>

    <!--VIRTUAL
    JOINT: Purpose: this element defines a virtual joint between a robot link and an external frame
    of reference (considered fixed with respect to the robot)-->
    <virtual_joint name="root_joint" type="floating" parent_frame="world_frame" child_link="base" />
</robot>